store/catalog.go

```diff
@@ -26,68 +29,176 @@ func NewCatalog(databaseID *int, store *Store) *Catalog {
-		return indexList[i].Position < indexList[j].Position
-	})
+	return &databaseData, nil
+}
+
+type schemaMap map[string]*catalog.Schema
 
-	var columnExpressions []string
-	for _, index := range indexList {
-		columnExpressions = append(columnExpressions, index.Expression)
+func (m schemaMap) getSchema(name string) *catalog.Schema {
+	if schema, found := m[name]; found {
+		return schema
 	}
 
-	return &catalog.Index{
-		Name:              indexList[0].Name,
-		TableName:         table.Name,
-		Type:              indexList[0].Type,
-		Unique:            indexList[0].Unique,
-		Primary:           indexList[0].Primary,
-		ColumnExpressions: columnExpressions,
-	}, nil
+	schema := &catalog.Schema{
+		Name: name,
+	}
+	m[name] = schema
+	return schema
 }
 
-// FindTable finds the table list by TableFind. Implement the catalog.Catalog interface.
-func (c *Catalog) FindTable(ctx context.Context, find *catalog.TableFind) ([]*catalog.Table, error) {
+func (c *Catalog) getSchemaList(ctx context.Context) ([]*catalog.Schema, error) {
+	schemaSet := make(schemaMap)
+
+	// find table list
 	tableList, err := c.store.FindTable(ctx, &api.TableFind{
 		DatabaseID: c.databaseID,
 	})
 	if err != nil {
 		return nil, err
 	}
-
-	var res []*catalog.Table
 	for _, table := range tableList {
-		if find.DatabaseName == table.Database.Name {
-			res = append(res, &catalog.Table{
-				Name:         table.Name,
-				DatabaseName: table.Database.Name,
-			})
+		tableData := convertTable(table)
+		schemaName := ""
+		if c.engineType == db.Postgres {
+			if schemaName, tableData.Name, err = splitPGSchema(tableData.Name); err != nil {
+				return nil, err
+			}
 		}
+		schema := schemaSet.getSchema(schemaName)
+		schema.TableList = append(schema.TableList, tableData)
 	}
 
-	return res, nil
+	// find view list
+	viewList, err := c.store.FindView(ctx, &api.ViewFind{
+		DatabaseID: c.databaseID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, view := range viewList {
+		viewData := convertView(view)
+		schemaName := ""
+		if c.engineType == db.Postgres {
+			if schemaName, viewData.Name, err = splitPGSchema(viewData.Name); err != nil {
+				return nil, err
+			}
+		}
+		schema := schemaSet.getSchema(schemaName)
+		schema.ViewList = append(schema.ViewList, viewData)
+	}
+
+	// find extension list
+	extensionList, err := c.store.FindDBExtension(ctx, &api.DBExtensionFind{
+		DatabaseID: c.databaseID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, extension := range extensionList {
+		schema := schemaSet.getSchema(extension.Schema)
+		schema.ExtensionList = append(schema.ExtensionList, &catalog.Extension{
+			Name:        extension.Name,
+			Version:     extension.Version,
+			Description: extension.Description,
+		})
+	}
+
+	var schemaList []*catalog.Schema
+	for _, schema := range schemaSet {
+		schemaList = append(schemaList, schema)
+	}
+	return schemaList, nil
+}
+
+func splitPGSchema(name string) (string, string, error) {
```

Comment:
I am wondering if we need to deal with pg specifically and just call it splitSchemaAndTableName?

For MySQL, there are 2 cases:

1. The table name is always without dot thus the schemaName is always empty.

2. If the MySQL table name could be `foo.bar` here, the split logic would incorrectly assign foo as the schema name. If that's the case, then would the current logic fail to deal with pg table name like `my_schema.foo.bar` where `foo.bar` is the table name?

Reply:
I think it's OK here. Because this function is only used in the store module, which means it only deals with the metadata in Bytebase. I think the metadata in Bytebase is trustworthy because it's defined by the Bytebase schema.

The root cause for this function is that Bytebase collects the PG table name as "schema.table" in https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/plugin/db/pg/sync.go?L190